core-bundle/composer.json

```
@@ -185,6 +185,7 @@
         "contao/manager-bundle": "^4.5 <4.5.2",
         "contao/manager-plugin": "<2.0 || >=3.0",
         "doctrine/cache": "<1.10",
+        "symfony/var-exporter": ">=8.0",
```

Shouldn't this be

```suggestion
        "symfony/var-exporter": ">=8@dev",
```

? Otherwise pre-releases are allowed.

---

Nobody uses pre-releases in production. And if someone explicitly wants to install the pre-release, we probably shouldn't prevent it because they most likely know what they are doing.

---

I don't agree with that. Also this conflict should be added to `contao/manager-bundle`, not the `contao/core-bundle`. If you use Contao 5.6 in a Symfony Application you are then free to still keep `symfony/var-exporter` in Version 8 and enable `doctrine.orm.enable_native_lazy_objects` instead. See also https://github.com/contao/contao/pull/9060

---

The unit tests fail without the conflict, so I guess it is correct here.

---

Since we don't directly require `symfony/var-exporter`, we could also solve this with a dev requirement for version 6 and 7 I guess. But until the underlying issue has been fixed, the conflict is the correct solution.

---

Yes, for the CI we should use a dev requirement in the `contao/core-bundle`. But for the issue in the application itself we should add the conflict only to the `contao/manager-bundle`.

---

I don't agree. Using the dev requirement only works by chance. For example, we cannot solve the `doctrine/dbal` problem in the same way, because there is a non-dev requirement already.

Additionally, having different version requirements for the same package in prod and dev is a bad compromise at best and will lead to more problems than it solves (such as "why does it not work in prod although all unit tests pass?" or vice versa).

IMHO the conflict is correct. And as long as we need to keep it, the `doctrine.orm.enable_native_lazy_objects` case is not possible with Contao (except with `"symfony/var-exporter": "8.0 as 7.4.999"`).

---

The best solution would be a `conflict-dev` in Composer, but that doesn't exist as far as I know, does it?

---

> IMHO the conflict is correct. And as long as we need to keep it, the doctrine.orm.enable_native_lazy_objects case is not possible with Contao (except with "symfony/var-exporter": "8.0 as 7.4.999").

But why would we force that? Contao itself does not care about any of this. It's an application configuration issue and thus needs to be handled by the `contao/manager-bundle`.

---

The unit tests fail without the conflict.

It does not affect us, because we run the tests in `contao/contao`, but everyone running the unit tests in `contao/core-bundle` will be affected. But I don't want to waist time discussing this, so changed in 2871c7172e2f5a3574e037993c3beb9b9c58245f.